BitwardenShared/UI/Platform/ExtensionSetup/ExtensionActivation/ExtensionActivationView.swift

```diff
@@ -8,87 +8,113 @@ struct ExtensionActivationView: View {
     // MARK: Properties
 
     /// The `Store` for this view.
-    @ObservedObject var store: Store<ExtensionActivationState, ExtensionActivationAction, Void>
+    @ObservedObject var store: Store<
+        ExtensionActivationState,
+        ExtensionActivationAction,
+        ExtensionActivationEffect
+    >
 
-    /// The title text in the view.
-    var title: String {
-        switch store.state.extensionType {
-        case .appExtension:
-            Localizations.extensionActivated
-        case .autofillExtension:
-            Localizations.autofillActivated
-        }
-    }
+    /// An action that opens URLs.
+    @Environment(\.openURL) private var openURL
 
-    /// The message text in the view.
-    var message: String {
-        switch store.state.extensionType {
-        case .appExtension:
-            Localizations.extensionSetup +
-                .newLine +
-                Localizations.extensionSetup2
-        case .autofillExtension:
-            Localizations.autofillSetup +
-                .newLine +
-                Localizations.autofillSetup2
+    // MARK: View
+
+    var body: some View {
+        Group {
+            if store.state.showLegacyView {
+                legacyContent
+            } else {
+                content
+            }
```

Comment:
⚠️ `legacyContent` handles two cases based on the `extensionType` in the state: 1) activating the autofill extension and 2) activating the action extension. These new changes should only affect activating the autofill extension (1). If you activate the action extension (app Settings > Auto-fill > App extension), you should still see the legacy content for the app extension, but you'll currently see the new changes for the autofill extension. Would it make sense to still show the legacy view for the app extension type?

Reply:
got this fixed up, great catch.